src/test/java/at/ac/tuwien/kr/alpha/common/depgraph/DependencyGraphTest.java

```diff
@@ -0,0 +1,310 @@
+package at.ac.tuwien.kr.alpha.common.depgraph;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.antlr.v4.runtime.CharStreams;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import at.ac.tuwien.kr.alpha.Alpha;
+import at.ac.tuwien.kr.alpha.common.Predicate;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.common.depgraph.io.DependencyGraphWriter;
+import at.ac.tuwien.kr.alpha.grounder.parser.ProgramParser;
+import at.ac.tuwien.kr.alpha.grounder.structure.ProgramAnalysis;
+
+public class DependencyGraphTest {
+
+	@SuppressWarnings("unused")
+	private static String generateRandomProgram(int numRules, int numPredicates, int maxRuleBodyLiterals) {
+		String[] predicates = new String[numPredicates];
+		for (int i = 0; i < predicates.length; i++) {
+			predicates[i] = "p" + Integer.toString(i + 1);
+		}
+
+		StringBuilder prgBuilder = new StringBuilder();
+		String tmpAtom;
+		int tmpBodyLiterals;
+		for (int i = 0; i < numRules; i++) {
+			tmpBodyLiterals = 1 + ((int) (Math.random() * maxRuleBodyLiterals));
+			tmpAtom = predicates[(int) (Math.random() * predicates.length)];
+			prgBuilder.append(tmpAtom).append(" :- ");
+			for (int j = 0; j < tmpBodyLiterals; j++) {
+				tmpAtom = predicates[(int) (Math.random() * predicates.length)];
+				prgBuilder.append(tmpAtom);
+				if (j < (tmpBodyLiterals - 1)) {
+					prgBuilder.append(", ");
+				}
+			}
+			prgBuilder.append(".");
+			prgBuilder.append("\n");
+		}
+		return prgBuilder.toString();
+	}
+
+	@Test
+	@Ignore("Not a real test, rather a playground for local testing while changing stuff")
```

Comment:
This does not look ideal. Please put the private/local testing stuff into a file that is not committed to git.